scrapy/extensions/feedexport.py

```diff
@@ -350,13 +350,19 @@ def close_spider(self, spider):
         return defer.DeferredList(deferred_list) if deferred_list else None
 
     def _close_slot(self, slot, spider):
+
+        def get_file(slot_):
+            if isinstance(slot_.file, PostProcessingManager):
+                return slot_.file.file
```

Comment:
After a quick glance at the failures, I am thinking we may need to close PostProcessingManager without closing the underlying file. Calling `slot_.close()` here before `return` may be enough.